main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1732,13 +1732,17 @@ object Typer {
           resultEff = Type.mkUnion(defaultEff :: ruleEffs, loc)
         } yield (resultCon, resultTyp, resultPur, resultEff)
 
-      case KindedAst.Expression.Spawn(exp, loc) =>
+      case KindedAst.Expression.Spawn(exp1, exp2, loc) =>
+        val regionVar = Type.freshVar(Kind.Bool, loc)
+        val regionType = Type.mkRegion(regionVar, loc)
         for {
-          (constrs, tpe, _, eff) <- visitExp(exp)
+          (constrs1, tpe1, _, eff1) <- visitExp(exp1)
+          (constrs2, tpe2, _, eff2) <- visitExp(exp2)
+          _ <- expectTypeM(expected = regionType, actual = tpe2, exp2.loc)
```

Comment:
This should just be unifyTypeM and not expectTypeM

Reply:
Can you help me understand this? Why is it `unifyTypeM` here and not, for example, in the `NewChanel` case above?

Reply:
They are functionally the same, but `expectTypeM` generates an error messages saying that "expected" was expected.

When working with type variables, it would be strange to get a message saying the type variable is expected, because it is not what the programmer wrote. Moreover, such unifications usually cannot fail. So its more of a stylistic choice reminding us of that fact.

Reply:
For example, we use `expectTypeM` for the condition of an if-then-else because we know we want a Boolean.